pkg/cloudprovider/azure.go

```diff
@@ -37,27 +38,38 @@ const (
 // to the Azure cloud API
 type Azure struct {
 	CloudProvider
-	resourceGroup        string
-	env                  azure.Environment
-	vmClient             compute.VirtualMachinesClient
-	virtualNetworkClient network.VirtualNetworksClient
-	networkClient        network.InterfacesClient
-	nodeMapLock          sync.Mutex
-	nodeLockMap          map[string]*sync.Mutex
+	resourceGroup                string
+	env                          azure.Environment
+	vmClient                     compute.VirtualMachinesClient
+	virtualNetworkClient         network.VirtualNetworksClient
+	networkClient                network.InterfacesClient
+	nodeMapLock                  sync.Mutex
+	nodeLockMap                  map[string]*sync.Mutex
+	azureWorkloadIdentityEnabled bool
 }
 
 func (a *Azure) initCredentials() error {
 	clientID, err := a.readSecretData("azure_client_id")
 	if err != nil {
-		return err
+		// Fallback to using client ID from env variable if not set.
+		clientID = os.Getenv("AZURE_CLIENT_ID")
+		if strings.TrimSpace(clientID) == "" {
```

Comment:
right now `clientID` doesn't have the spaces trimmed in it's value, does it matter?

Reply:
I copied this pattern from [here](https://github.com/openshift/cluster-ingress-operator/pull/906/commits/b37da5aedc32985e675955a2686482a621c72e8a#diff-5ed822890c33068b8a84c26cfb283868438d555942dd45a235e8a5a9b81de15eR52); I think the intent is to protect against malformed environment variables.  I could remove it if you'd like, but I don't see any harm in leaving it how it is